### PR TITLE
 Make find-doc-nits check for newly added undocumented symbols

### PR DESCRIPTION
**Pull request by mattcaswell**
We create lists of undocumented functions and macros as they are now so that find-doc-nits can check for newly introduced functions/macros that are undocumented.

This works in a similar way to the -u and -d options to find-doc-nits. These count undocumented symbols and print a detailed list of undocumented symbols repsectively. This PR adds the -v and -e options to restrict the count/detailed list to newly added undocumented symbols only.

There is also a new -s option that does the same as -e except that it produces no output if there are no newly undocumented symbols.

We also amend "make doc-nits" to add the -s option which should cause travis to fail if a PR adds undocumented symbols.

There is also the new option -o to amend the behaviour of -v or -e to check symbols that were newly
added since the release of 1.1.1.

Finally this PR fixes a documentation error that was found during development of this PR (a symbol that existed in 2 man pages), and an error in the definition of a macro that caused a false report from find-doc-nits.

### CODE REVIEW

**Comment by mspncp** [comment]
> There is also the new option -o to amend the behaviour of -v or -e to check symbols that were newly added since the release of 1.1.1.

Just a thought: using `git blame`, the script could even spit out the culprit who introduced it ;-)

**Comment by mattcaswell** [comment]
> Just a thought: using git blame, the script could even spit out the culprit who introduced it ;-)

Not sure I want to do automatic finger pointing at this stage :-)

**Comment by mattcaswell** [comment]
Ping?

File: util/find-doc-nits [hunk 1]
```diff
@@ -503,14 +545,20 @@ sub checkflags() {
...
+$opt_e = 1 if $opt_s;
+$opt_v = 1 if $opt_o || $opt_e;
 
-die "Need one of -[cdlnpu] flags.\n"
-    unless $opt_c or $opt_l or $opt_n or $opt_u;
+die "Cannot use both -u and -v" if $opt_u && $opt_v;
+die "Cannot use both -d and -e" if $opt_d && $opt_e;
+
+die "Need one of -[cdelnpuv] flags.\n"
+    unless $opt_c or $opt_l or $opt_n or $opt_u or $opt_v;
```
Comment (levitte) [inline util/find-doc-nits]: Maybe the text should match the check?

Comment (mattcaswell) [reply util/find-doc-nits]: Options `d`, `e`, `s`, `o` and `p` all imply one of the other options so we don't need to check them explicitly. I added a comment about that. Also the text was actually missing a couple of options so I added that too.

**Comment by mattcaswell** [comment]
> Also, what's the plan for updating the missing files? Any method to do so?

It's not particularly critical. Probably something we can just do on an occasional basis as we see fit. If a missing function gets documented, it doesn't matter if it is in one of the "missing" files (no error is reported). The worst that could happen would be that a missing function gets documented, and then at some later point the documentation is accidentally deleted. In that case find-doc-nits would not spot the sudden removal of the documentation.

We *could* make find-doc-nits complain if it finds a documented function in its undocumented list. That way if you document a function you also have to remove it from the missing list (otherwise travis "make doc-nits" builds will fail). Not sure if it is worth it at this stage.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Comment by mspncp** [comment]
Backported to 1.1.1 via #10094.

**Pull request closed without merging**